### ISSUE TITLE
linter: new jsx-a11y rule to confirm `<svg>` has an accessible name

### ISSUE THREAD

**Issue by mehm8128** [oxc-project/oxc#22115]
related to https://github.com/oxc-project/oxc/issues/1141

There is an issue to propose a rule to add new rule to confirm that `<svg>` has an accessible name on `eslint-plugin-jsx-a11y`: [New rule proposal around SVGs · Issue #733 · jsx-eslint/eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/733)

And Biome has introduced such a rule: [noSvgWithoutTitle | Biome](https://biomejs.dev/linter/rules/no-svg-without-title/)

Oxlint should also implement similar rule. We can regard this rule as a svg-version of [jsx-a11y/alt-text | The JavaScript Oxidation Compiler](https://oxc.rs/docs/guide/usage/linter/rules/jsx_a11y/alt-text.html).

We should support two methods to provide an accessible name to `<svg>` element.
- to use `<title>` element as the first child of `<svg>`
- to mark `<svg>` as `role="img"` and provide `aria-label` or `aria-labedledby` attribute to it

For example (cited from Biome document)

valid:
```html
<svg>
    <title>Pass</title>
    <circle />
</svg>

<svg role="img" aria-labelledby="title">
    <span id="title">Pass</span>
</svg>

<svg role="img" aria-label="title">
    <span id="title">Pass</span>
</svg>

<svg role="graphics-symbol">
    <title>Pass</title>
    <rect />
</svg>

<svg aria-hidden="true"><rect /></svg>

<svg role="img" aria-label="">
    <span id="">Pass</span>
</svg>

<svg role="presentation">foo</svg>
```

invalid:
```html
<svg>foo</svg>

<svg>
    <title></title>
    <circle />
</svg>

<svg>
    <rect />
    <rect />
    <g>
        <title>foo</title>
        <circle />
        <circle />
    </g>
</svg>
```

**Comment by mehm8128** [comment]
If there are no specific concerns about introducing this rule, I will create a draft PR.
